### PR TITLE
Transform propagator and compute at map out of lowering.

### PR DESCRIPTION
**Pull request by csarofeen**
Make a traversal pass to transform the fusion consistent with a tensor.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/lower2device.cpp [hunk 1]
```diff
@@ -356,7 +356,6 @@ kir::Expr* GpuLower::lowerExpr(const Expr* expr) {
 }
 
 GpuLower* GpuLower::current() {
-  TORCH_INTERNAL_ASSERT(active_gpu_lower != nullptr);
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/lower2device.cpp:364]: Is this reasonable, or should we keep this assert and send an argument to the compute at map whether or not its a lowering map?

**Comment by csarofeen** [comment]
Note: https://github.com/csarofeen/pytorch/pull/676/commits/95a03712b0d4f28c76dac333b78a0fe100005d00

File: torch/csrc/jit/codegen/cuda/compute_at_map.h [hunk 2]
```diff
@@ -71,6 +71,8 @@ class TORCH_CUDA_CU_API ComputeAtMap {
   std::string toString();
 
  private:
+  bool has_lowered_kir_ = false;
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/compute_at_map.h:78]: I assume the intention is to make `ComputeAtMap` available without lowering. Wouldn't it be simpler and cleaner if the mappings for KIR nodes are completely separated from the mappings for the Fusion IR? For example, how about letting `ComputeAtMap` only take care of the Fusion IR nodes, and create, say, class `kir::ComputeAtMap`, that is "lowered" from `ComputeAtMap`?

Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/compute_at_map.h:78]: @csarofeen  I updated the interface just a little bit in order to make things more explicit. `build` takes `Fusion*` and `GpuLower*` as its parameters. If `GpuLower*` is `nullptr`, it doesn't generate the KIR version of the mappings.

https://github.com/csarofeen/pytorch/pull/688/commits/94ee1ecf9664ed44f764074a141553f34bca80bf

**Review by naoyam — APPROVED** [review]

**Pull request merged**